src/main/scala/io/iohk/ethereum/jsonrpc/server/http/JsonRpcHttpsServer.scala

```diff
@@ -51,84 +36,11 @@ class JsonRpcHttpsServer(
           case Success(serverBinding) => log.info(s"JSON RPC HTTPS server listening on ${serverBinding.localAddress}")
           case Failure(ex) => log.error("Cannot start JSON HTTPS RPC server", ex)
         }
-      case Left(error) => log.error(s"Cannot start JSON HTTPS RPC server due to: $error")
-    }
-  }
-
-  /**
-    * Constructs the SSL context given a certificate
-    *
-    * @param certificateKeyStorePath, path to the keystore where the certificate is stored
-    * @param password for accessing the keystore with the certificate
-    * @return the SSL context with the obtained certificate or an error if any happened
-    */
-  private def obtainSSLContext(
-      certificateKeyStorePath: String,
-      certificateKeyStoreType: String,
-      password: String
-  ): HttpsSetupResult[SSLContext] = {
-    val passwordCharArray: Array[Char] = password.toCharArray
-
-    val maybeKeyStore: HttpsSetupResult[KeyStore] = Try(KeyStore.getInstance(certificateKeyStoreType)).toOption
-      .toRight(s"Certificate keystore invalid type set: $certificateKeyStoreType")
-    val keyStoreInitResult: HttpsSetupResult[KeyStore] = maybeKeyStore.flatMap { keyStore =>
-      val keyStoreFileCreationResult = Option(new FileInputStream(certificateKeyStorePath))
-        .toRight("Certificate keystore file creation failed")
-      keyStoreFileCreationResult.flatMap { keyStoreFile =>
-        Try(keyStore.load(keyStoreFile, passwordCharArray)) match {
-          case Success(_) => Right(keyStore)
-          case Failure(err) => Left(err.getMessage)
-        }
-      }
-    }
-
-    keyStoreInitResult.map { ks =>
-      val keyManagerFactory: KeyManagerFactory = KeyManagerFactory.getInstance("SunX509")
-      keyManagerFactory.init(ks, passwordCharArray)
-
-      val tmf: TrustManagerFactory = TrustManagerFactory.getInstance("SunX509")
-      tmf.init(ks)
-
-      val sslContext: SSLContext = SSLContext.getInstance("TLS")
-      sslContext.init(keyManagerFactory.getKeyManagers, tmf.getTrustManagers, secureRandom)
-      sslContext
+      case Left(error) =>
+        log.error(s"Cannot start JSON HTTPS RPC server due to: $error")
+        throw new RuntimeException(error.reason)
```

Comment:
I prefer to use `IllegalStateException` is more adequate here

Reply:
Done